tests/tracer/test_settings.py

```diff
@@ -320,3 +322,24 @@ def test_environment_header_tags():
     assert config._header_tag_name("User-agent") == "http.user_agent"
     # Case insensitive
     assert config._header_tag_name("User-Agent") == "http.user_agent"
+
+
+@contextmanager
+def no_exceptions_raised():
+    yield
+
+
+@pytest.mark.parametrize(
+    "env,expected,raises",
+    (
+        (dict(), (128, True), no_exceptions_raised()),
+        (dict(DD_TRACE_X_DATADOG_TAGS_MAX_LENGTH="0"), (0, False), no_exceptions_raised()),
+        (dict(DD_TRACE_X_DATADOG_TAGS_MAX_LENGTH="256"), (256, True), no_exceptions_raised()),
+        (dict(DD_TRACE_X_DATADOG_TAGS_MAX_LENGTH="-1"), None, pytest.raises(ValueError)),
```

Comment:
Should there be a test for an extremely large int and other defaults to see whether it throws an error or not? 

Example values: `1000` and `512`?